tools/audit_console_messages.py

```
@@ -42,25 +42,50 @@
+    an EVEN number (including zero) means it is not -- e.g. ^^& is a literal caret followed by
+    an ACTIVE '&', not an escaped one. A single-character regex lookbehind cannot express this
+    for an arbitrary run of carets, so it is checked by explicit backward scan instead."""
+    carets = 0
+    i = pos - 1
+    while i >= 0 and text[i] == '^':
+        carets += 1
+        i -= 1
+    return carets % 2 == 1
+
+
+def _first_unescaped(pattern: re.Pattern, text: str):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `_first_unescaped`

(ANN202)